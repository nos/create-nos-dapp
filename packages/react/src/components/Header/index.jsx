import React from "react";
import injectSheet from "react-jss";
import PropTypes from "prop-types";

import SpinningLogo from "./../SpinningLogo";

const styles = {
  header: {
    backgroundColor: "#f0f0f0",
    color: "#333333",
    padding: "24px",
    marginBottom: "32px"
  },
  title: {
    fontSize: "1.5em"
  }
};

const Header = ({ classes, title }) => (
  <header className={classes.header}>
    <h1 className={classes.title}>{title}</h1>
    <SpinningLogo />
  </header>
);

Header.propTypes = {
  classes: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired
};

export default injectSheet(styles)(Header);
