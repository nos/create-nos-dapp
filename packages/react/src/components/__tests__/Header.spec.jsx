import React from "react";
import { render } from "enzyme";

import TestPropTypes from "./../../__helpers__/TestPropTypes.helper";

import Header from "./../Header";

describe("Components", () => {
  describe("Header", () => {
    test("should render", () => {
      const wrapper = render(<Header title="Hello world!" />);
      expect(wrapper).toMatchSnapshot();
    });

    test("should validate props", () => {
      TestPropTypes(<Header title="Hello world!" />, 0);
      TestPropTypes(<Header />, 1);
    });
  });
});
