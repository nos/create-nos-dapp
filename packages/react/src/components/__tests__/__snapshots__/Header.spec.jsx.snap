// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Header should render 1`] = `
<header
  class="Header-header-0-1-1"
>
  <h1
    class="Header-title-0-1-2"
  >
    Hello world!
  </h1>
  <div
    class="SpinningLogo-logoContainer-0-1-3"
  >
    <img
      alt="nOS logo"
      class="SpinningLogo-nosLogo-0-1-4"
      src="test-file-stub"
    />
  </div>
</header>
`;
